packages/mobile/src/identity/verification.ts

```diff
@@ -193,7 +191,35 @@ export function* startVerification({ withoutRevealing }: StartVerificationAction
   yield put(refreshAllBalances())
 }
 
+export function* restartableVerification(initialWithoutRevealing: boolean) {
+  let isRestarted = false
+  while (true) {
+    const withoutRevealing = !isRestarted && initialWithoutRevealing
+    yield call(navigate, Screens.VerificationLoadingScreen, {
```

Comment:
Are there any negative repercussions to calling `navigate` in a loop like this? What happens to the screen stack if `VerificationLoadingScreen` is already at the top when it's called? I'm not sure but maybe `replace` is more appropriate in that instance?

Reply:
As far as I understand, if you call navigate with the current screen it would not result into any modification to the stack. If this is not true, we should re-think our approach here https://github.com/celo-org/celo-monorepo/blob/22cd8449fe457101badcde844c292b0763ee4c8e/packages/mobile/src/import/saga.ts#L64 as well, where we navigate to the same screen just with different params.

Reply:
If I am understanding [this thread](https://github.com/react-navigation/react-navigation/issues/2882) correctly, it looks like this method will continue to push the same screen onto the top of the stack.

If so, this probably has negative implications on performance as well as creating some unwanted consequences of breaking the "back" carat. Maybe you can use the `setParams` method instead?

Reply:
From here https://reactnavigation.org/docs/navigating/#navigate-to-a-route-multiple-times

> ... The navigate function roughly means "go to this screen", and if you are already on that screen then it makes sense that it would do nothing.

Also, in our case we will navigate to the same screen only if we are stuck on `VerificationLoadingScreen`.

From the same page:

> Each time you call push we add a new route to the navigation stack. When you call navigate it first tries to find an existing route with that name, and only pushes a new route if there isn't yet one on the stack.

So, I assume it does not add to the stack already existing routes when you navigate to them.

Lmk if that makes sense for you?

Reply:
My understanding based on [this comment in the thread](https://github.com/react-navigation/react-navigation/issues/2882#issuecomment-656466559) was that if you're passing new params, then it would create a new screen.

To be honest, I'm not too sure about this either way. If you didn't notice any adverse effects when you were testing, I am ok with merging as is given we do this in other parts of the code base.

Reply:
So, I've logged `route.key` when navigating to the route with `{withoutRevealing: true}` and then again with `{withoutRevealing: false}` and key stays the same. I assume it does not create a new screen, but I might be wrong...